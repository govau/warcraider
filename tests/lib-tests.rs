#[cfg(test)]
mod tests {
    use insta::{assert_debug_snapshot, assert_json_snapshot};

    #[test]
    fn test_html_parser_snapshots() {
        let raw_html = std::fs::read_to_string("tests/moneysmart.htm").unwrap();
        let html: warcraider::HTMLResult = warcraider::find_html_parser(1, 1,
"https://www.moneysmart.gov.au/life-events-and-you/life-events/divorce-and-separation/divorce-and-separation-financial-checklist", &raw_html);
        assert_json_snapshot!( html,{
        ".meta_tags" => "[meta_tags]"
        });
        let mut meta_tags = html
            .meta_tags
            .iter()
            .map(|(k, v)| format!("{}:{}", k, v))
            .collect::<Vec<String>>();
        meta_tags.sort();
        assert_debug_snapshot!(meta_tags);
    }

    #[test]
    fn test_make_urls_absolute() {
        let result = warcraider::make_urls_absolute(
            "http://example.com",
            vec![
                String::from("/index.htm"),
                String::from("http://google.com"),
            ],
        );
        //dbg!(&result);
        assert_eq!(
            result,
            ["http://example.com/index.htm", "http://google.com/"]
        );
    }
    #[test]
    fn test_make_urls_absolute_trailing_quote() {
        let result = warcraider::make_urls_absolute(
            "http://example.com",
            vec![
                String::from("/index.htm"),
                String::from("/index.htm'"),
                String::from("/index.htm\""),
                String::from("/index.htm&quot;"),
                String::from("/index.htm%20"),
                String::from("/notindex.htm"),
            ],
        );
        //dbg!(&result);
        assert!(
            result
                == vec![
                    "http://example.com/index.htm",
                    "http://example.com/notindex.htm"
                ]
        );
    }
}
