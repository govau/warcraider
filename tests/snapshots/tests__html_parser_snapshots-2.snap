---
source: tests/lib-tests.rs
expression: meta_tags
---
[
    "Description:The divorce and separation financial checklist gives you some practical steps to separate your finances and get your money on track. Visit ASIC\'s MoneySmart website for more. ",
    "content-type:text/html; charset=utf-8",
    "dcterms.creator:c=au;o=Australian Government;ou=Australian Government Australian Securities & Investments Commission",
    "dcterms.date:2019-04-26",
    "dcterms.description:The divorce and separation financial checklist gives you some practical steps to separate your finances and get your money on track. Visit ASIC\'s MoneySmart website for more. ",
    "dcterms.identifier:/life-events-and-you/life-events/divorce-and-separation/divorce-and-separation-financial-checklist",
    "dcterms.publisher:c=au;o=Australian Government;ou=Australian Government Australian Securities & Investments Commission",
    "dcterms.subject:Consumer protection; EF Economics and Finance",
    "dcterms.title:Divorce and separation financial checklist",
    "og:description:The divorce and separation financial checklist gives you some practical steps to separate your finances and get your money on track. Visit ASIC\'s MoneySmart website for more. ",
    "og:title:Divorce and separation financial checklist | ASIC\'s MoneySmart",
    "og:type:website",
    "og:url:https://www.moneysmart.gov.au/life-events-and-you/life-events/divorce-and-separation/divorce-and-separation-financial-checklist",
    "position:5",
    "theme-color:#6b8e06",
    "twitter:card:summary",
    "twitter:description:The divorce and separation financial checklist gives you some practical steps to separate your finances and get your money on track. Visit ASIC\'s MoneySmart website for more. ",
    "twitter:site:@moneysmartteam",
    "twitter:title:Divorce and separation financial checklist | ASIC\'s MoneySmart",
    "viewport:width=device-width, initial-scale=1",
]
